crates/iceberg/src/metadata_columns.rs

```diff
@@ -37,40 +35,52 @@ pub const RESERVED_FIELD_ID_FILE: i32 = i32::MAX - 1;
 /// Reserved column name for the file path metadata column
 pub const RESERVED_COL_NAME_FILE: &str = "_file";
 
-/// Lazy-initialized Arrow Field definition for the _file metadata column.
-/// Uses Run-End Encoding for memory efficiency.
-static FILE_FIELD: Lazy<Arc<Field>> = Lazy::new(|| {
-    let run_ends_field = Arc::new(Field::new("run_ends", DataType::Int32, false));
-    let values_field = Arc::new(Field::new("values", DataType::Utf8, true));
-    Arc::new(
-        Field::new(
-            RESERVED_COL_NAME_FILE,
-            DataType::RunEndEncoded(run_ends_field, values_field),
-            false,
-        )
-        .with_metadata(HashMap::from([(
-            PARQUET_FIELD_ID_META_KEY.to_string(),
-            RESERVED_FIELD_ID_FILE.to_string(),
-        )])),
-    )
+/// Lazy-initialized Iceberg field definition for the _file metadata column.
+/// This field represents the file path as a required string field.
+static FILE_FIELD: Lazy<NestedFieldRef> = Lazy::new(|| {
+    Arc::new(NestedField::required(
+        RESERVED_FIELD_ID_FILE,
+        RESERVED_COL_NAME_FILE,
+        Type::Primitive(PrimitiveType::String),
+    ))
 });
 
-/// Returns the Arrow Field definition for the _file metadata column.
+/// Returns the Iceberg field definition for the _file metadata column.
 ///
 /// # Returns
-/// A reference to the _file field definition (RunEndEncoded type)
-pub fn file_field() -> &'static Arc<Field> {
+/// A reference to the _file field definition as an Iceberg NestedField
+pub fn file_field() -> &'static NestedFieldRef {
     &FILE_FIELD
 }
 
-/// Returns the Arrow Field definition for a metadata field ID.
+/// Extracts the primitive type from a metadata field.
+///
+/// # Arguments
+/// * `field` - The metadata field
+///
+/// # Returns
+/// The PrimitiveType of the field, or an error if the field is not a primitive type
+pub fn metadata_field_primitive_type(field: &NestedFieldRef) -> Result<PrimitiveType> {
```

Comment:
I don't think we should add this method for metadata column only. If you think this is necessary, we could add a `as_primitive_type_result` method in `Type` .